game_board.gd

```diff
@@ -204,22 +209,25 @@ func _select_active_unit(cell: Vector2) -> void:
 	# registered in the `cell`.
 	if not _units.has(cell):
 		return
-
-	if _units[cell].isPlayerControllable:
+#todo: fix previously freed error on next line when clicking on tile after the occupant dies
+	if is_occupied(cell) && _units[cell].isPlayerControllable && _units[cell].actionPoints > 0:
+		print("can control")
 		# When selecting a unit, we turn on the overlay and path drawing. We could use signals on the
 		# unit itself to do so, but that would split the logic between several files without a big
 		# maintenance benefit and we'd need to pass extra data to the unit.
 		# I decided to group everything in the GameBoard class because it keeps all the selection logic
 		# in one place. I find it easy to keep track of what the class does this way.
 		_active_unit = _units[cell]
 		_unit_base = _units[cell].base
-		_active_unit.isSelected = true
+		_active_unit.set_is_selected(true)
```

Comment:
unit should probably be the one with the methods that change it's stats or status